imagebuildah/stage_executor.go

```diff
@@ -1168,6 +1193,22 @@ func (s *StageExecutor) Execute(ctx context.Context, base string) (imgID string,
 			if err != nil {
 				return "", nil, fmt.Errorf("error checking if cached image exists from a previous build: %w", err)
 			}
+			// All the best effort to find cache on localstorage have failed try pulling
+			// cache from remote repo if `--cache-from` was configured.
+			if cacheID == "" && s.executor.cacheFrom != "" && cacheKey != "" {
```

Comment:
`cacheKey` should never be empty if `s.executor.cacheFrom` is set, so no need to check for it here.